src/Command/RestartCommand.php

```diff
@@ -67,12 +67,17 @@ public function execute(InputInterface $input, OutputInterface $output)
                 $input->getOption('type'),
                 $input->getOption('max-attempts')
             );
-        }
 
-        if (0 === \count($messages)) {
-            $output->writeln('Nothing to restart, bye.');
+            /*
+             * Check messages count only for not pulling mode
+             * to avoid PHP warning message
+             * since ErroneousMessageIterator does not implement Countable.
+             */
+            if (0 === \count($messages)) {
```

Comment:
Or maybe change this to `$messages instanceof \Countable && 0 === \count($messages)`? That way you do not need this lengthy explanation. Also, maybe you could use http://php.net/manual/en/iterator.valid.php in the case of the iterator? It does not seem to have side effects: https://3v4l.org/1rjEc

Reply:
In our case $messages var can be either an array or ErroneousMessageIterator, array is not instance of \Countable and we need to check count of $messages only on array type case, we don't need to create one more if statement, so i have just moved the check of the count in the array case bloc